cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -247,4 +265,21 @@ public WorkflowToken deepCopy() {
     }
     return builder.build();
   }
+
+  private void ensureEnoughMemoryAvailable(String key, Value value) {
+    int totalSizeAfterAdding  = Bytes.toBytes(key).length + Bytes.toBytes(value.toString()).length + currentSizeBytes;
```

Comment:
Technically it can overflow. It's better be:

```java
int bytesLeft; /// Initialize to max size in constructor
....
int left = bytesLeft;
left -= key.length;
if (left <= 0) {
  // throw exception
}
left -= value.length;
if (left <= 0) {
  // throw exception
}
// Update the size as well
bytesLeft = left;
```